### PR TITLE
Configure mclag ports - b - #65

### PR DESCRIPTION
**Pull request by vivek-metricdust**
backend fix for [#65](https://github.com/STORDIS/orca_nw_lib/issues/65)

Functionality which will suport the deletion of single member (Portchannel) of mclag
if the member are provided it will delete only those portchannels else it will delete all the portchannels form members.
Respective test cases are added to check the deletion of a single member and also delete all the members.

There is if statement in the mclag.py which will check 

- if members is provided as key and with portchannel names as values it will remove only those portchannels.
- if member keys is provided with empty array it will remove all the members 
- if there is no member key then it will delete the mclag

### CODE REVIEW

File: network/test/test_mclag.py [hunk 1]
```diff
@@ -44,6 +44,24 @@ def test_mclag_config(self):
...
+        }
+
+        response = self.put_req("device_mclag_list", request_body)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        response = self.get_req(
+            "device_mclag_list", {"mgt_ip": device_ip_1, "domain_id": self.domain_id}
+        )
+
+        self.assertEqual(response.json().get("domain_id"), self.domain_id)
```
Comment (kamalkrbh) [inline network/test/test_mclag.py:62]: @vivek-metricdust, also assert session_timeout

File: network/test/test_mclag.py [hunk 2]
```diff
@@ -280,6 +298,64 @@ def test_mclag_member_config(self):
...
         self.perform_add_port_chnl(request_body)
 
+        # create mclag and add member
+        request_body_members = {
+            "mgt_ip": device_ip_1,
+            "domain_id": self.domain_id,
+            "mclag_members": [self.mem_port_chnl, self.mem_port_chnl_2],
+        }
+
+        response = self.del_req("device_mclag_list", request_body_members)
```
Comment (kamalkrbh) [inline network/test/test_mclag.py:308]: @vivek-metricdust , instead use remove_mclag function from test_common.py as done other tests above.

File: network/test/test_mclag.py [hunk 3]
```diff
@@ -280,6 +298,64 @@ def test_mclag_member_config(self):
...
+        self.assertTrue(
+            response.status_code == status.HTTP_200_OK
+            or any(
+                "resource not found" in res.get("message", "").lower()
+                for res in response.json()["result"]
+                if res != "\n"
+            )
+        )
+        response = self.put_req("device_mclag_list", request_body_members)
```
Comment (kamalkrbh) [inline network/test/test_mclag.py:317]: @vivek-metricdust, assert properties of mclag after put request.

File: network/test/test_mclag.py [hunk 4]
```diff
@@ -280,6 +298,64 @@ def test_mclag_member_config(self):
...
+
+        # check deletion of single member
+        request_body_member = {
+            "mgt_ip": device_ip_1,
+            "domain_id": self.domain_id,
+            "mclag_members": [self.mem_port_chnl],
+        }
+
+        response = self.del_req("device_mclag_list", request_body_member)
```
Comment (kamalkrbh) [inline network/test/test_mclag.py:326]: @vivek-metricdust, use remove_mclag function.

File: network/test/test_mclag.py [hunk 5]
```diff
@@ -280,6 +298,64 @@ def test_mclag_member_config(self):
...
+        self.assertTrue(response.status_code == status.HTTP_200_OK)
+
+        response = self.get_req("device_mclag_list", request_body_members)
+        self.assertEqual(response.json().get("mclag_members")[0], self.mem_port_chnl_2)
```
Comment (kamalkrbh) [inline network/test/test_mclag.py:330]: @vivek-metricdust , As here only members are expected to be deleted also assert that mclag still exists with it properties.

File: network/test/test_mclag.py [hunk 6]
```diff
@@ -280,6 +298,64 @@ def test_mclag_member_config(self):
...
+
+        # check deletion of all members with empty member array
+        request_body_member = {
+            "mgt_ip": device_ip_1,
+            "domain_id": self.domain_id,
+            "mclag_members": [],
```
Comment (kamalkrbh) [inline network/test/test_mclag.py:336]: @vivek-metricdust , In such a case mclag should be deleted as the mclag array is empty. For deleting all mclag members array should be filled with will all members name.

File: network/test/test_mclag.py [hunk 7]
```diff
@@ -280,6 +298,64 @@ def test_mclag_member_config(self):
...
+        }
+        response = self.del_req("device_mclag_list", request_body_member)
+        self.assertTrue(
```
Comment (kamalkrbh) [inline network/test/test_mclag.py:339]: @vivek-metricdust, Not only http status but also non existence of mclag or members should be asserted.

File: network/test/test_mclag.py (hunk 5 above)
Comment (vivek-metricdust) [reply network/test/test_mclag.py:330]: this is test case to check the new delete api which deletes the particular member so have used the delete request

File: network/mclag.py [hunk 8]
```diff
@@ -80,7 +85,8 @@ def device_mclag_list(request):
             # to be for deleting the MCLAG
             if mclag_members:
                 try:
-                    del_mclag_member(device_ip)
+                    for mem in mclag_members or []:
```
Comment (kamalkrbh) [inline network/mclag.py:88]: @vivek-metricdust, Better have a separate end point for deleting member ports of an mclag, and deletion of mclag means deletion of mclag domain, if mclag has got member interfaces and fives error while deleting let it show to user and let user decide. In that case user will have to remove mclag member ports first and then actually delete the MCLAG domain.

File: network/test/test_mclag.py (hunk 2 above)
Comment (vivek-metricdust) [reply network/test/test_mclag.py:308]: This is the test for checking the delete member end point. But the URL for end point has been changed so will update accordingly

File: network/test/test_mclag.py (hunk 1 above)
Comment (vivek-metricdust) [reply network/test/test_mclag.py:62]: assertion for session_timeout added.

File: network/mclag.py (hunk 8 above)
Comment (vivek-metricdust) [reply network/mclag.py]: fixes in [2691f44](https://github.com/STORDIS/orca_backend/pull/49/commits/2691f44918f477845927db2fa3a0906ab326534a)

File: network/test/test_mclag.py (hunk 7 above)
Comment (vivek-metricdust) [reply network/test/test_mclag.py]: fixes in [2691f44](https://github.com/STORDIS/orca_backend/pull/49/commits/2691f44918f477845927db2fa3a0906ab326534a)

File: network/test/test_mclag.py (hunk 6 above)
Comment (vivek-metricdust) [reply network/test/test_mclag.py]: fixes in [2691f44](https://github.com/STORDIS/orca_backend/pull/49/commits/2691f44918f477845927db2fa3a0906ab326534a)

File: network/test/test_mclag.py (hunk 5 above)
Comment (vivek-metricdust) [reply network/test/test_mclag.py:333]: fixes in [2691f44](https://github.com/STORDIS/orca_backend/pull/49/commits/2691f44918f477845927db2fa3a0906ab326534a)

File: network/test/test_mclag.py (hunk 3 above)
Comment (vivek-metricdust) [reply network/test/test_mclag.py:312]: fixes in [2691f44](https://github.com/STORDIS/orca_backend/pull/49/commits/2691f44918f477845927db2fa3a0906ab326534a)

**Comment by vivek-metricdust** [comment]
All retested changes are added in commit fixes in [2691f44](https://github.com/STORDIS/orca_backend/pull/49/commits/2691f44918f477845927db2fa3a0906ab326534a)

New end point for deletion of member added.
additional test cases to update the port channel member also added.

File: network/test/test_mclag.py [hunk 9]
```diff
@@ -295,19 +301,64 @@ def test_mclag_member_config(self):
...
+        }
+        
+        response = self.del_req("delete_mclag_members", request_body_delete_members)
+        self.assertTrue(response.status_code == status.HTTP_200_OK)
 
-        # cleanup members
-        response = self.del_req("device_mclag_list", request_body_members)
-        print("---", response)
+        response = self.get_req("device_mclag_list", request_body_delete_members)
+        self.assertEqual(response.json().get("domain_id"), request_body_delete_members['domain_id'])
```
Comment (kamalkrbh) [inline network/test/test_mclag.py:361]: @vivek-metricdust , Also get the mclag member ports and assert them to be empty or None.

Comment (vivek-metricdust) [reply network/test/test_mclag.py:361]: assertion added fixes in [55d5b36](https://github.com/STORDIS/orca_backend/pull/49/commits/55d5b3615541e16fd6a693aca8244a313be1a636)

**Pull request merged**